desktop/ui/src/main/java/org/datacleaner/output/datastore/DatastoreCreationDelegateImpl.java

```diff
@@ -29,23 +30,27 @@
  */
 public class DatastoreCreationDelegateImpl implements DatastoreCreationDelegate {
 
-	private static final Logger logger = LoggerFactory.getLogger(DatastoreCreationDelegateImpl.class);
+    private static final Logger logger = LoggerFactory.getLogger(DatastoreCreationDelegateImpl.class);
 
-	private final MutableDatastoreCatalog _datastoreCatalog;
+    private final DatastoreCatalog _datastoreCatalog;
 
-	public DatastoreCreationDelegateImpl(MutableDatastoreCatalog datastoreCatalog) {
-		_datastoreCatalog = datastoreCatalog;
-	}
+    public DatastoreCreationDelegateImpl(DatastoreCatalog datastoreCatalog) {
+        _datastoreCatalog = datastoreCatalog;
+    }
 
-	@Override
-	public void createDatastore(Datastore datastore) {
-		final String name = datastore.getName();
-		synchronized (_datastoreCatalog) {
-			if (_datastoreCatalog.containsDatastore(name)) {
-				logger.warn("Datastore '{}' already exists. No new datastore will be created!", name);
-			} else {
-				_datastoreCatalog.addDatastore(datastore);
-			}
-		}
-	}
+    @Override
+    public void createDatastore(Datastore datastore) {
+        final String name = datastore.getName();
+        synchronized (_datastoreCatalog) {
+            if (_datastoreCatalog.containsDatastore(name)) {
+                logger.warn("Datastore '{}' already exists. No new datastore will be created!", name);
+            } else {
+                if (_datastoreCatalog instanceof MutableDatastoreCatalog) {
+                    ((MutableDatastoreCatalog) _datastoreCatalog).addDatastore(datastore);
+                } else {
+                    logger.error("Tried to create datastore '{}', but the datastore catalog is not mutable", name);
```

Comment:
To the user, nothing really happens, as far as I can see. Shouldn't this throw an exception so the user can get a warning?